clients/src/test/java/org/apache/kafka/test/TestUtils.java

```diff
@@ -563,7 +563,7 @@ public static <T extends Throwable> void assertFutureThrows(
         String expectedMessage
     ) {
         T receivedException = assertFutureThrows(future, expectedCauseClassApiException);
-        assertEquals(expectedMessage, receivedException.getMessage());
+        assertTrue(receivedException.getMessage().contains(expectedMessage));
```

Comment:
In gerenal, I perfer to explicit message since we my use simple error message when developing.
Here is a example, https://github.com/apache/kafka/pull/17954/commits/0bf1c9be0a0f2094a09ef7f56528ac582257ab58#r1860502292.

If we use `contain`, the error will be hidden.

Reply:
Yes, it's better to make it deterministic, so avoid using `contains` in a commonly used util method, let's alter the failed message to it's full version, @peterxcli

Reply:
Hi @TaiJuWu @dengziming, thanks for your advice, fixed! PTAL